lib/clevertap.js

```diff
@@ -373,6 +374,423 @@ CleverTap.prototype.events = function(query, options, callback) {
     });
 };
 
+/*Delete Api
+ Deletes a user profile from the CleverTap system.
+ */
+CleverTap.prototype.delete = function(query, options, callback) {
+//Validate the query object for the "profiles" endpoint
+    var error = _validate("profiles", query);
+
+    if (typeof(options) === 'function' || !options) {
+        callback = options;
+        options = {};
+    }
+
+    return new Promise( (resolve, reject) => {
+        if (error) {
+            console.error(error);
+            if (callback && typeof callback === 'function') {
+                callback(null);
+            }
+            reject(error);
+            return;
+        }
+
+        this.api.profile(query, options, (res) => {
```

Comment:
looks like `this.api.delete` should be here. And we have to send the correct endpoint as a first argument